notifications_utils/recipients.py

```diff
@@ -321,9 +321,16 @@ def _get_error_for_field(self, key, value):  # noqa: C901
                     email_address.validate_email_address(value)
                 if self.template_type == "sms":
                     if self.allow_sms_to_uk_landline:
```

Comment:
I think we don't need this if/else statement here now?

We should always do what's in the `if`, and never what's in the `else`, right?

Reply:
Currently we can do both. We will be removing the `phone_number.validate_phone_number` function in the next ticket. So in the long run, yes. But the scope of this PR is just adding the allow_landline flag. (In practice we could probably just remove the branching here and just use the `if` ... probably won't break anything, but I think that's beyond the scope of this PR).

Reply:
Hmmm - I can see how it could be out of scope - but since it's just removing an if statement, and most likely it will not break anything, isn't it more efficient to do it here? If you feel that will muddy the waters for you though - that's fair, it's ok to do it in a separate PR.